### PR TITLE
Delayed io

### PR DESCRIPTION
**Pull request by ilyavf**
Can delay socket connection now. Added documentation.

### CODE REVIEW

File: delay-io.js [hunk 1]
```diff
@@ -0,0 +1,190 @@
+/**
+ * @function steal-socket.delay-io delay-io
```
Comment (matthewp) [inline delay-io.js]: This is internal, right? If so add this to this top section:

```
@hide
```

File: ignore-zone.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+/**
+ * @function steal-socket.ignore-zone ignore-zone
```
Comment (matthewp) [inline ignore-zone.js]: Same as above, this needs an `@hide`

File: delay-io.js (hunk 1 above)
Comment (ilyavf) [inline delay-io.js]: I was thinking about this as a public. You can import only this function if you want just to delay socket.io on client.

File: ignore-zone.js (hunk 2 above)
Comment (ilyavf) [inline ignore-zone.js]: Do you want may be to move the delay-io examples to the main doc page of steal-socket and remove the child docs hiding both `ignore-zone` and `ignore-ssr`?

**Comment by matthewp** [comment]
@ilyavf looks good now, let me know when this is ready

**Comment by ilyavf** [comment]
Hey @matthewp , I think this is ready. I also changed the version to 4.0.0 if its OK with you.

**Comment by matthewp** [comment]
Usually we do that when releasing and not in the PR, just so you know in the future.

**Pull request merged**